aspnetcore/blazor/test.md

```diff
@@ -0,0 +1,132 @@
+
+The following table summarizes the difference between the two testing approaches.
+
+| Capability                       | E2E testing                             | Unit testing                     |
+| -------------------------------- | --------------------------------------- | -------------------------------- |
+| Test scope                       | Blazor component (Razor/C#) with CSS/JS | Blazor component (Razor/C#) only |
+| Test execution time              | Seconds                                 | Milliseconds                     |
+| Access to the component instance | No                                      | Yes                              |
+| Sensitive to the environment     | Yes                                     | No                               |
+
+## Choose the most appropriate test approach
+
+Consider the scenario when choosing the type of testing to perform. Some considerations are described in the following table.
+
+| Scenario | Suggested approach | Remarks |
+| -------- | ------------------ | ------- |
+| Component without JS interop logic | Unit testing | When there's no dependency on JS interop in a Blazor component, the component can be tested without access to JS or the DOM API. In this scenario, there are no disadvantages to choosing unit testing. |
+| Component with simple JS interop logic | Unit testing | It's common for components to query the DOM or trigger animations through JS interop. Unit testing is usually preferred in this scenario, since it's straightforward to mock the JS interaction through the <xref:Microsoft.JSInterop.IJSRuntime> interface. |
+| Component that depends on complex JS code | Unit testing and separate JS testing | If a component uses JS interop to call a large or complex JS library but the interaction between the Blazor component and JS library is simple, then the best approach is likely to treat the component and JS library or code as two separate parts and test each individually. Test the Blazor component with a unit testing library, and test the JS with a JS testing library. |
+| Component with logic that depends on JS manipulation of the browser DOM | E2E testing | When a component's functionality is dependent on JS and its manipulation of the DOM, verify both the JS and Blazor code together in an E2E test. This is the approach that the Blazor framework developers have taken with Blazor's browser rendering logic, which has tightly-coupled C# and JS code. The C# and JS code must work together to correctly render Blazor components in a browser.
+
+## Test components with bUnit
+
+There's no official Microsoft testing framework for Blazor, but the community-driven project [bUnit](https://github.com/egil/bUnit) provides a convenient way to unit test Blazor components.
+
+> [!NOTE]
+> bUnit is a third-party testing library and isn't supported or maintained by Microsoft or the .NET Foundation.
```

Comment:
My plan is to submit bunit to the .net foundation, but I guess that is not relevant at this point?

Reply:
I'll make it just say "Microsoft."